PWGHF/TableProducer/candidateSelectorDstar.cxx

```diff
@@ -0,0 +1,349 @@
+  /// @tparam T1 Table iterator type of D0 Candidate
+  /// @tparam T2 Table iterator type of Dstar Candidate
+  /// @param d0candidate D0 candidate object
+  /// @param dstarCandidate Dstar candidate object
+  /// @return true/false depending on success of selection
+  template <typename T1, typename T2>
+  bool selectionTopolConjugate(const T1& d0candidate, const T2& dstarCandidate)
+  {
+    auto dstarpT = dstarCandidate.pt();
+    auto pTBin = findBin(binsPtForDstar, dstarpT);
+    if (pTBin == -1) {
+      return false;
+    }
+    auto SoftPi = dstarCandidate.template prongPi_as<TracksSel>();
+
+    if (SoftPi.sign() > 0.) { // Selection of D*+
+      // auto Mpipik = std::array{massPi,massPi,massK};
+      // auto InvDstar = dstarCandidate.dstarInvMass(Mpipik);
+      auto InvDstar = hfHelper.invMassDstar(dstarCandidate);
+
+      // auto Mpik = std::array{massPi,massK};
+      // auto InvD0 = d0candidate.d0m(Mpik);
+      auto InvD0 = hfHelper.invMassD0ToPiK(d0candidate);
+
+      // LOGF(info,"+ %d,%f,%f",SoftPi.sign(),InvDstar,InvD0);
+      if (std::abs(InvDstar - InvD0) > cutsForDstar->get(pTBin, "DeltaMDstar")) {
+        return false;
+      }
+      // cut on D0 daughter DCA - need to add secondary vertex constraint here
+      if (std::abs(d0candidate.impactParameter0()) > cutsForD0->get(pTBin, "d0pi") || std::abs(d0candidate.impactParameter1()) > cutsForD0->get(pTBin, "d0K")) {
+        return false;
+      }
+      // cut on cos(theta*)
+      if (std::abs(hfHelper.cosThetaStarD0(d0candidate)) > cutsForD0->get(pTBin, "cos theta*")) {
+        return false;
+      }
+
+    } else if (SoftPi.sign() < 0) { // Selection of D*-
+      // auto Mpikpi = std::array{massPi,massK,massPi};
+      // auto InvDstar = dstarCandidate.dstarInvMass(Mpikpi);
+      auto InvDstar = hfHelper.invMassAntiDstar(dstarCandidate);
+
+      // auto Mkpi = std::array{massK,massPi};
+      // auto InvD0 = d0candidate.d0m(Mkpi);
+      auto InvD0 = hfHelper.invMassD0barToKPi(d0candidate);
+
+      // LOGF(info,"- %d,%f,%f",SoftPi.sign(),InvDstar,InvD0);
+      if (std::abs(InvDstar - InvD0) > cutsForDstar->get(pTBin, "DeltaMDstar")) {
+        return false;
+      }
+      // cut on D0 daughter DCA - need to add secondary vertex constraint here
+      if (std::abs(d0candidate.impactParameter0()) > cutsForD0->get(pTBin, "d0K") || std::abs(d0candidate.impactParameter1()) > cutsForD0->get(pTBin, "d0pi")) {
+        return false;
+      }
+      // cut on cos(theta*)
+      if (std::abs(hfHelper.cosThetaStarD0bar(d0candidate)) > cutsForD0->get(pTBin, "cos theta*")) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  void process(TracksSel const&, aod::Hf2Prongs const&, aod::HfD0fromDstar const& rowsD0cand, aod::HfDstarCand const& rowsDstarCand)
+  {
+    LOG(info) << "selector: processQA called";
+    for (auto& DstarCand : rowsDstarCand) {
+      auto dstarIndex = DstarCand.globalIndex();
+      auto D0Cand = rowsD0cand.iteratorAt(dstarIndex);
```

Comment:
Why don't we join the two tables?

Reply:
Earlier, when we did not have a separate ```HfD0FromDstarBase``` table, It was not possible because of many deceiving names for columns in the ```HfCand2ProngBase``` table. But now I think we can join these two.

Reply:
Not joining them has some advantages when writing selection functions in the selector class. We can write separate template selection functions for D0 and Dstar to make the code more clean and easily understandable for any reader.